.github/workflows/config.yml

```diff
@@ -90,35 +77,51 @@ jobs:
           steps.cache-msys2.outputs.cache-hit != 'true'
         shell: python scripts\msys-bash.py {0}
         run: |
-           scripts/list-build-dependencies.sh -m msys2 -c clang ${{ matrix.conf.without_packages }} | xargs pacman -S --noconfirm
+           pacman -S --noconfirm \
+             meson \
+             mingw-w64-x86_64-ccache \
+             mingw-w64-x86_64-fluidsynth \
+             mingw-w64-x86_64-gcc \
+             mingw-w64-x86_64-libpng \
+             mingw-w64-x86_64-ncurses \
+             mingw-w64-x86_64-opusfile \
+             mingw-w64-x86_64-pkg-config \
+             mingw-w64-x86_64-SDL2 \
+             mingw-w64-x86_64-SDL2_net \
+             mingw-w64-x86_64-zlib
            .github/scripts/shrink-msys2.sh
 
-      - name:  Install dependencies ${{ matrix.conf.without_packages }} (Linux)
+      - name:  Install dependencies (Linux)
         if:    matrix.system.name == 'Linux'
         run: |
           sudo apt-get -y update
-          sudo apt-get install -y $(./scripts/list-build-dependencies.sh -m apt -c gcc ${{ matrix.conf.without_packages }})
+          sudo apt-get install $(cat ./.github/packages/ubuntu-20.04-apt.txt)
 
-      - name:  Install dependencies ${{ matrix.conf.without_packages }} (macOS)
+      - name:  Install dependencies (macOS)
         if:    matrix.system.name == 'macOS'
-        run:   brew install $(./scripts/list-build-dependencies.sh -m brew -c clang ${{ matrix.conf.without_packages }})
+        run: |
+          brew install meson $(cat ./.github/packages/macos-brew.txt)
 
       - name:  Build ${{ matrix.conf.configure_flags }} (Windows)
         if:    matrix.system.name == 'Windows'
         shell: python scripts\msys-bash.py {0}
-        run: >
-          scripts/build.sh -c ${{ matrix.system.compiler }} -t debug --bin-path /mingw64/bin
-          ${{ matrix.conf.configure_flags }} ${{ matrix.system.missing_packages }}
+        run: |
+          export PATH="/mingw64/bin:$PATH"
```

Comment:
This PATH override is such an annoyance in mingw; so many things are broken without it (and also had to handle it in the prior build script).  I guess this is still the easiest option; although would be nice if meson allowed setting PATH, which we could do when cygwin/msys2 is detected.

Reply:
Meson does not have a built-in option for overriding path *and I think it's a good thing*. There's `meson setup --bindir=name`, but it's changing the bin directory for installation (not overriding path).

Meson-way of doing this is preparation of [native file](https://mesonbuild.com/Release-notes-for-0-49-0.html#native-config-files) - we already have two such native files in directory `.github/meson/` - perhaps I will add a third one for MSYS2. I tried that once already and failed, but I probably made a mistake somewhere.